### DISCUSSION TITLE
Performance Issues and Browser Crashes with Large Forms in Vueform (500+ Fields)

### DISCUSSION THREAD

**Discussion by alahanoun** [vueform/vueform#351] [Questions]
I’m experiencing very slow loading times and browser crashes when using vueform for a large form with over 500 fields. Loading this many fields at once is causing performance issues and is crashing the browser. The form also includes many select elements with numerous options, which may be adding to the performance issues.

Is there any way to improve performance with large schemas like this?

**Comment by adamberecz** [comment]
One way to improve this could be to remove unnecessary parts (like before/after texts) of the [`ElementLayout`](https://github.com/vueform/vueform/blob/main/themes/blank/templates/ElementLayout.vue) and overriding it as a [custom template](https://vueform.com/docs/styles-and-layout#replacing-templates). You might also consider breaking them into [steps](https://vueform.com/docs/breaking-forms-into-steps#using-form-steps) or [tabs](https://vueform.com/docs/breaking-forms-into-steps#using-tabs) or implementing a [loader](https://vueform.com/recipes/show-form-loader-for-large-forms).
